osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -446,7 +446,31 @@ Task IMultiplayerClient.HostChanged(int userId)
 
         Task IMultiplayerClient.SettingsChanged(MultiplayerRoomSettings newSettings)
         {
-            Scheduler.Add(() => updateLocalRoomSettings(newSettings));
+            Debug.Assert(APIRoom != null);
+            Debug.Assert(Room != null);
+
+            Scheduler.Add(() =>
+            {
+                // ensure the new selected item is populated immediately.
+                var playlistItem = APIRoom.Playlist.SingleOrDefault(p => p.ID == newSettings.PlaylistItemId);
+
+                if (playlistItem != null)
```

Comment:
Two things that bother me slightly here:

* Not sure if the outer schedule is even necessary anymore?
* If `playlistItem` ever happens to be null then `updateLocalRoomSettings()` won't get called at all (which wasn't the case before). That should theoretically never happen, I'd like to think, so maybe best to upgrade that to a `Single()` and handle the null case as catastrophic failure?

Reply:
The outer schedule was specifically added as smoogi mentioned that we shouldn't be accessing `APIRoom` properties from a non-update thread (the `.SingleOrDefault` call in this case).

Reply:
That covers the outer schedule part of my question, but what about the playlist item being potentially null? Is that something that can ever happen?

Reply:
Second point needs some further investigation. I'm not sure if it can be null, but does require one of the two proposed changes to remain correct.

Reply:
I've changed the call to `.Single()`, seems to work as expected.